src/modules/mavlink/mavlink_main.h

```diff
@@ -498,6 +499,14 @@ class Mavlink
 	 */
 	struct ping_statistics_s &get_ping_statistics() { return _ping_stats; }
 
+	/**
+	 * Fills two 3D covariance matrixes (3x3 + 3x3) from a 6D cross covariance matrix URT (21 values).
+	 * The first matrix corresponds to the position or the linear velocity covariance,
+	 * while the second corresponds to the attitude or the angular velocity covariance
+	 */
+	void covariance_from_matrixurt_helper(float urt[21], matrix::SquareMatrix3f &matrix1,
```

Comment:
This is weirdly specific for mavlink. If it deserves its own function you should put it in Matrix or somewhere in PX4 lib/.

Reply:
You are right - and I actually need this to be different: the idea is to actually have a 6x6 rotation matrix instead of 3x3 blocks.